ClientTester/MultiplayerClient.cs

```diff
@@ -1,83 +1,93 @@
-using NitroxClient.Map;
+using NitroxModel.Core;
 using NitroxModel.Logger;
+using NitroxModel.MultiplayerSession;
 using NitroxModel.Packets;
 using NitroxModel.Packets.Processors.Abstract;
-using System;
-using System.Collections.Generic;
-using System.Threading;
 using UnityEngine;
 
 namespace ClientTester
 {
     public class MultiplayerClient
     {
-        public IPacketSender PacketSender { get; }
-        public ClientBridge ClientBridge { get; }
-        public Logic Logic { get; }
-        public object PacketProcessorsByType { get; private set; }
+        private DeferringPacketReceiver packetReceiver { get; }
+        private IMultiplayerSession multiplayerSession { get; }
 
         public Vector3 ClientPos = new Vector3(-50f, -2f, -38f);
 
-        private readonly VisibleCells visibleCells;
-        private readonly DeferringPacketReceiver packetReceiver;
-        private readonly TcpClient client;
         private readonly string playerName;
 
         public MultiplayerClient(string playerId)
         {
             Log.SetLevel(Log.LogLevel.ConsoleInfo | Log.LogLevel.ConsoleDebug);
             playerName = playerId;
-            visibleCells = new VisibleCells();
-            packetReceiver = new DeferringPacketReceiver(visibleCells);
-            client = new TcpClient(packetReceiver);
-            ClientBridge = new ClientBridge(client);
-            PacketSender = ClientBridge;
-            Logic = new Logic(ClientBridge, visibleCells, packetReceiver);
+
+            NitroxServiceLocator.InitializeDependencyContainer(new ClientAutoFaqRegistrar());
+            packetReceiver = NitroxServiceLocator.LocateService<DeferringPacketReceiver>();
+            multiplayerSession = NitroxServiceLocator.LocateService<IMultiplayerSession>();
         }
 
         public void Start(string ip)
         {
-            ClientBridge.Connect(ip, playerName);
+            Dictionary<Type, PacketProcessor> packetProcessorMap = GeneratePacketProcessorMap();
+            multiplayerSession.ConnectionStateChanged += ConnectionStateChangedHandler;
+            multiplayerSession.Connect(ip);
 
-            var iterations = 0;
-            do
+            for (int iterations = 0; iterations < 20; iterations++)
             {
                 Thread.Sleep(250);
-                ProcessPackets();
+                ProcessPackets(packetProcessorMap);
+            }
+        }
 
-                iterations++;
-                if (iterations >= 20)
-                {
-                    break;
-                }
-            } while (ClientBridge.CurrentState == ClientBridgeState.WaitingForRerservation);
+        private Dictionary<Type, PacketProcessor> GeneratePacketProcessorMap()
+        {
+            Dictionary<Type, PacketProcessor> packetProcessorMap = new Dictionary<Type, PacketProcessor>
+            {
+                { typeof(MultiplayerSessionPolicy), new MultiplayerSessionPolicyProcessor(multiplayerSession) },
```

Comment:
Can we infer the packet type here? It's only two items now, but this procedure is error prone nevertheless. Especially if someone changes the type of packet handled by the processor later on (very unlikely though).

```cs
Dictionary<Type, PacketProcessor> packetProcessorMap = new List<PacketProcessor>
{
    new MultiplayerSessionPolicyProcessor(multiplayerSession),
    new MultiplayerSessionReservationProcessor(multiplayerSession)
}.ToDictionary(k => k.GetType().BaseType.GetGenericArguments()[0], v => v);
```

Reply:
Fair enough. I know killing the client tester has been discussed so did not want to spend a lot of time here, but a quick fix to make it less brittle is cool with me.

Reply:
I've been going through the PR on GitHub - missed/misread the fact that this is `ClientTester`. Nice to fix for consistency, but if we kill it I'm happy as well.